device/tt_soc_descriptor.h

```diff
@@ -167,31 +157,19 @@ class tt_SocDescriptor {
         coordinate_manager.reset(new CoordinateManager(*other.coordinate_manager));
```

Comment:
This is a mess, the problem is that we don't really even want to copy SocDescriptor in my opinion, it is just a case at the moment because of some legacy copying in Cluster. This will be deleted. Until then I can change it to shared pointer